constrained-voucher.mkd

```diff
@@ -646,36 +646,52 @@ INSERT_FIG_FROM_FILE examples/voucher-example1.txt END
 Support for ECDSA with sha256 (secp256k1 and prime256v1 curves) is REQUIRED.
 Support for EdDSA is encouraged. [TBD EDNOTE: Expand and add a reference why. ]
 
-An example of the supported COSE-sign1 object structure is shown in {{fig-cose}}.
+An example of the supported COSE_Sign1 object structure is shown in {{fig-cose}}.
 
 ~~~~
 COSE_Sign1(
   [
     h'A101382E',        # { "alg": ES256K }
     {
-      "kid" : h'7890....1234'  # hash256(public key)
+      4 : h'7890A03F1234'  # "kid" binary key identifier
     },
-    h'1234....5678', #voucher-request binary content
-    h'4567....1234', #voucher-request binary public signature
+    h'1234....5678', #voucher-request binary content (CBOR)
+    h'4567....1234'  #voucher-request binary public signature
   ]
 )
 ~~~~
-{: #fig-cose title='cose-sign1 example in CBOR diagnostic notation' align="left"}
-
-The {{COSE-registry}} specifies the integers that replace the strings and the mnemonics in {{fig-cose}}.
-The value of the key identifier "kid" parameter is an example value.
-Usually a hash of the public key is used to identify the public key. The choice of key identifier method is
-vendor-specific.
-The public key and its hash are derived from the relevant certificate (Pledge, Registrar or MASA certificate).
-[TBD EDNOTE: how can MASA know which kid method the Registrar has used/supports? Does it matter?]
-
-In {{cosesign}} a binary cose-sign1 object is shown based on the voucher-request example of {{example2}}.
+{: #fig-cose title='COSE_Sign1 example in CBOR diagnostic notation' align="left"}
+
+The {{COSE-registry}} specifies the integers/encoding for the "alg" and "kid" fields in {{fig-cose}}. The "alg" 
+field restricts the key usage for verification of this COSE object to a particular cryptographic algorithm.
+
+The "kid" field is optionally present: it is an unprotected field that identifies the public key of the key pair that was used to sign this 
+message. The value of the key identifier "kid" parameter is an example value.
+Usually a hash of the public key is used to identify the public key, but a device serial number may also be used. The choice of key identifier method is
+vendor-specific. If "kid" is not present, then a verifying party needs to use other context information to 
+retrieve the right public key to verify the COSE_Sign1 object against. For example, this context information 
+may be a unique serial number encoded in the binary content (CBOR) field.
+
+A Registrar MAY use a "kid" parameter in its RVR to identify its signing key as used to sign the RVR. 
+The method of generating this "kid" is vendor-specific and SHOULD be configurable in the Registrar to 
+support commonly used methods. A Registrar SHOULD have a configuration option to set the "kid" to a particular 
+binary value. By default, a Registrar SHOULD NOT include a "kid" parameter in its RVR since the signing key 
+is already identified by the included signing certificates in the COSE "x5bag" structure.
```

Comment:
```suggestion
support commonly used methods. 
In order to support future business cases and supply chain integrations, a Registrar MUST be configurable, on a per-manufacturer basis, to be able to configure the "kid" to a particular value. 
Both binary and string values are to be supported, each being inserted using a CBOR bstr or tstr.
By default, a Registrar does not include a "kid" parameter in its RVR since the signing key 
is already identified by the included signing certificates in the COSE "x5bag" structure.
```

I know that MUST always feels like a large (Mjöllnir class) hammer best reserved for unique cases.
But, what I've learnt from IESG reviews is that every SHOULD has to come with the exceptions listed.